src/ergw_aaa_diameter.erl

```diff
@@ -150,6 +152,17 @@ validate_transport(Opt, Value) ->
 validate_transport_error(Opt, Value) ->
     throw({error, {options, {Opt, Value}}}).
 
+validate_handler(Opts) ->
```

Comment:
```suggestion
validate_term_cause_mapping(Opt, Value)
  when is_atom(Opt), is_integer(Value) ->
    Value;
validate_term_cause_mapping(Opt, Value) ->
    throw({error, {termination_cause_mapping, {Opt, Value}}}).

validate_term_cause_mapping(Opts) ->
    ergw_aaa_config:validate_options(
      fun validate_term_cause_mapping/2, Opts, ?DEFAULT_TERMINATION_CAUSE_MAP, map).
```

Note: this change termination_cause_mapping ot a **map**, code using id needs to be adjusted.

Reply:
Thanks your example